### DISCUSSION TITLE
Are STLB, ITLB and DTLB separate structures in the code?

### DISCUSSION THREAD

**Discussion by Setu-Gupta** [ChampSim/ChampSim#444] [General]
I have a couple of questions regarding STLB, DTLB and ITLB in ChampSim.

1. I assume that ITLB is the instruction TLB, and DTLB is the TLB for data accesses. Is this correct or am I misinterpreting the meaning of these terms?
2. If my interpretation of DTLB and ITLB is correct, then what is STLB?
3. I assume that ChampSim models STLB, DTLB and ITLB as structures which are separate from the cache. However, I'm observing that insertions made into STLB are being evicted from the L1D cache. If these structures are separate from the caches, how is this possible? Is this a bug or am I misinterpreting how the TLBs are modeled?

**Comment by ngober** [comment]
The STLB is a unified second-level TLB for both instructions and data. The TLBs are all modeled with the same model as the data caches (`CACHE`), but with different parameters to make them work on pages, rather than blocks. There is no inclusivity between any of the caches or TLBs. See #350 for more work on that.
